src/librustc_session/session.rs

```diff
@@ -542,7 +542,7 @@ impl Session {
     }
 
     pub fn no_landing_pads(&self) -> bool {
-        self.opts.debugging_opts.no_landing_pads || self.panic_strategy() == PanicStrategy::Abort
+        self.panic_strategy() == PanicStrategy::Abort
     }
```

Comment:
I wonder if this should be inlined or not.

Reply:
Won't LTO take care of it regardless?

Reply:
I don't mean in terms of performance, only code organization.
The reason these methods exist is because the conditions they check are non-trivial, otherwise options should be checked directly.

Reply:
> I don't mean in terms of performance, only code organization.
> The reason these methods exist is because the conditions they check are non-trivial, otherwise options should be checked directly.

I don't personally agree with eddyb's end assertion; sometimes checking options directly makes sense, but I have seen cases where it can be easier to read the overall logic with helper methods, especially if the names of the helper methods end up encoding something that would otherwise be written as a comment alongside the call site.

Having said that, I don't care enough one way or the other in this case to try to argue that this helper method should stay in place. I just don't want the comment here to, on its own, be treated as precedent for a general rule across the board.